src/shogun/mathematics/linalg/LinalgNamespace.h

```diff
@@ -106,17 +112,35 @@ LinalgBackendBase* infer_backend(const Container<T>& a, const Container<T>& b)
  * and a shallow-copy of SGVector with vector on CPU if GPU backend not available
  */
 template <typename T>
-SGVector<T> to_gpu(const SGVector<T>& vector)
+void to_gpu(SGVector<T>& a, SGVector<T>& b)
 {
```

Comment:
it's one thing that datastructures are not thread safe... but this is something that has to otherwise things gonna be really fuckedup...